enhancements/metering/slim-install.md

```diff
@@ -0,0 +1,210 @@
+
+#### Extending the `StorageLocation` CRD to Support Postgresql
+
+A `StorageLocation` custom resource is responsible for abstracting away details about the underlying metering storage backend, exposing the minimal configuration required to show the end-user where metering data has been persisted. Currently, this object only supports Hive databases and therefore needs to be able to replicate the same functionality we provide with Hive, but with Postgresql interfaces.
+
+When the metering slim installation has been enabled, the default `StorageLocation` custom resource that gets reconciled by the metering-ansible-operator should attempt to make this `CREATE DATABASE metering ...` database call using the Postgres go-client, and update it's status once that database has been created.
+
+Note: Postgresql has no native support for creating a database if it doesn't already exist, so the following call will need to be made instead:
+
+```SQL
+SELECT 'CREATE DATABASE metering' WHERE NOT EXISTS (SELECT FROM pg_database WHERE datname = 'metering')
+```
+
+#### Outlining a Potential Migration Path
+
+There's currently an undocumented PUSH API for the reporting-operator to statically inject Prometheus data that matches a specific JSON format. This API is largely used in metering end-to-end testing scenarios and is likely, not robust or stable enough to rely on for a full data set migration from existing metering installations. In theory, a migration effort is technically possible, but a more robust API implementation would be needed.
+
+### Risks and Mitigations
+
+- Need to support TLS and authentication for communication between the reporting-operator and the Postgres database instance.
+- Users wanting to use an existing Postgres database instance for metering usage.
+
+## Design Details
+
+### Test Plan
+
+- TBD
+
+### Graduation Criteria
+
+- Provide a data migration path for existing metering installations.
+- Sufficient time for feedback.
+- Sufficient testing in the following areas: upgrade, downgrade, scale, e2e.
+
+### Upgrade / Downgrade Strategy
+
+- TBD
+
+### Version Skew Strategy
+
+- N/A
+
+## Implementation History
+
+- N/A
+
+## Drawbacks
+
+- Continue to utilize the existing codebase for periodically importing metrics from Prometheus still relies on the averaged or estimated vector values that get returned by the Prometheus query_range API endpoint.
```

Comment:
I don't think this is the case, the averaging or estimation part. https://prometheus.io/docs/prometheus/latest/querying/api/#range-queries You spec a range and Prom gives you back the exact values it has as discrete elements in a vector.

Reply:
Yeah, the current phrasing might need some work. What I was trying to convey is that you're still relying on having to specify a time resolution when hitting the query_range API endpoint, so in the case that the step size is fairly large, you're potentially dealing with more skewed results as you're sample less frequently.

Reply:
Again, just closing loop from an out of band conversation, there _is_ some averaging going on with the query_range API, and larger step sizes would make the effect more noticeable.